### PR TITLE
Remove remaining bloc dependencies

### PR DESCRIPTION
**Pull request by lockieRichter**
Also updoot README

### CODE REVIEW

File: lib/src/providers/repository.dart [hunk 1]
```diff
@@ -1,12 +1,37 @@
...
 import 'package:riverpod_annotation/riverpod_annotation.dart';
 
 part 'repository.g.dart';
 
 @riverpod
-BookRepository bookRepository(BookRepositoryRef ref) => FirebaseBookRepository(
-      ref.read(firebaseAuthProvider),
-      ref.read(firebaseDatabaseProvider),
-    );
+class BookRepository extends _$BookRepository {
```
Comment (shockbytes) [inline lib/src/providers/repository.dart:10]: I don't fully understand from what we extend here.

File: lib/src/providers/repository.dart [hunk 2]
```diff
@@ -1,12 +1,37 @@
...
+    return _addBook(book, BookState.READING);
+  }
+
+  Future<void> addToRead(Book book) {
+    return _addBook(book, BookState.READ);
+  }
+
+  Future<void> _addBook(Book book, BookState bookState) {
+    final Book updatedBook = book.copyWith(newState: bookState);
+    return state.create(updatedBook);
```
Comment (shockbytes) [inline lib/src/providers/repository.dart:35]: Where is this `state` coming from?

File: lib/src/providers/repository.dart [hunk 3]
```diff
@@ -1,12 +1,37 @@
...
+    return _addBook(book, BookState.READ_LATER);
+  }
+
+  Future<void> addToReading(Book book) {
+    return _addBook(book, BookState.READING);
+  }
+
+  Future<void> addToRead(Book book) {
+    return _addBook(book, BookState.READ);
+  }
```
Comment (shockbytes) [inline lib/src/providers/repository.dart:31]: These are all methods added manually. Why do we need the `build()` method then here at all?

File: lib/src/providers/repository.dart (hunk 1 above)
Comment (lockieRichter) [reply lib/src/providers/repository.dart:10]: Annotating a class by `@riverpod` defines a new shared state accessible using the generated provider. Part of the generation is creating this class that we are extending. See https://github.com/rrousselGit/riverpod/tree/master/examples/counter for a basic example of creating a riverpod generated class.

File: lib/src/providers/repository.dart (hunk 2 above)
Comment (lockieRichter) [reply lib/src/providers/repository.dart:35]: This is the state that is initialised in the Build() method, in this case the state is the `FirebaseBookRepository`.

File: lib/src/providers/repository.dart (hunk 3 above)
Comment (lockieRichter) [reply lib/src/providers/repository.dart:31]: The build method initialises the state (which is the `FirebaseBookRepository`)

**Review by shockbytes — APPROVED** [review]
Thanks for the explanation and for taking care of updating the docs too!

**Pull request merged**